src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -336,7 +336,17 @@ private void DualModeConnect_AcceptAsync_Helper(IPAddress listenOn, IPAddress co
                 Assert.NotNull(clientSocket);
                 Assert.True(clientSocket.Connected);
                 Assert.Equal(AddressFamily.InterNetworkV6, clientSocket.AddressFamily);
-                Assert.Equal(connectTo.MapToIPv6(), ((IPEndPoint)clientSocket.LocalEndPoint).Address);
+                if (connectTo == IPAddress.Loopback)
+                {
+                    // Allow the IPv4 Loopback to be returned in place of the IPv4 mapped-to-IPv6 Loopback.
+                    IPAddress IPv4LoopbackInIPv6Format = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 127, 0, 0, 1 }, 0);
+                    Assert.True(IPAddress.Loopback.MapToIPv6().Equals(((IPEndPoint)clientSocket.LocalEndPoint).Address) ||
+                        IPv4LoopbackInIPv6Format.Equals(((IPEndPoint)clientSocket.LocalEndPoint).Address));
```

Comment:
Yeah that may be better. Though we're only getting ```::ffff:127.0.0.``` and ```::127.0.0.1``` here, ```::1``` would also be a valid loopback.